File: python/course-repo/solution/01-skip-the-workflow/src/webhooks/send_via_workflow.py
"""Fire the same webhook delivery wrapped in a Workflow - the comparison shape."""

import asyncio
import sys

from temporalio.client import Client

from .shared import ECHO_SERVER_URL, TASK_QUEUE, WebhookDelivery
from .workflow import WebhookWorkflow


async def main(event_id: str) -> None:
    client = await Client.connect("localhost:7233")
    result = await client.execute_workflow(
        WebhookWorkflow.run,
        args=[WebhookDelivery(
            url=ECHO_SERVER_URL,
            payload={"event_id": event_id, "type": "order.created", "amount": 99.99},
            event_id=event_id,
        )],
        id=f"wf-{event_id}",
        task_queue=TASK_QUEUE,
    )
    print(f"Workflow completed with activity returning status {result}")


if __name__ == "__main__":
    event_id = sys.argv[1] if len(sys.argv) > 1 else "evt_002"
    asyncio.run(main(event_id))

File: java/course-repo/exercise/06-same-code-runs-anywhere/src/main/java/webhook/SendViaWorkflow.java
package webhook;

import io.temporal.client.WorkflowClient;
import io.temporal.client.WorkflowOptions;
import io.temporal.serviceclient.WorkflowServiceStubs;
import java.util.Map;

public class SendViaWorkflow {
    public static void main(String[] args) {
        String eventId = args.length > 0 ? args[0] : "evt_002";

        WorkflowServiceStubs service = WorkflowServiceStubs.newLocalServiceStubs();
        WorkflowClient client = WorkflowClient.newInstance(service);

        WebhookDelivery req = new WebhookDelivery(
                Shared.WEBHOOK_RECEIVER_URL,
                Map.of("eventId", eventId, "type", "order.created", "amount", 99.99),
                eventId);

        // Same Activity, submitted as a Workflow step instead of directly.
        WebhookWorkflow workflow = client.newWorkflowStub(
                WebhookWorkflow.class,
                WorkflowOptions.newBuilder()
                        .setWorkflowId("wf-" + eventId)
                        .setTaskQueue(Shared.TASK_QUEUE)
                        .build());

        int status = workflow.run(req);
        System.out.println("Workflow completed with Activity returning status " + status);
    }
}
